File: python/jzoffer-42.py
# -*- coding:utf-8 -*-
class Solution:
    def FindNumbersWithSum(self, array, tsum):
        # write code here
        res = []
        i = 0
        j = len(array) - 1
        while i < j:
            if array[i] + array[j] == tsum:
                res.append(array[i])
                res.append(array[j])
                break
            while i < j and array[i] + array[j] > tsum:
                j -= 1
            while i < j and array[i] + array[j] < tsum:
                i += 1
        return res

File: java/jzoffer-42.java
import java.util.ArrayList;
public class Solution {
    public ArrayList<Integer> FindNumbersWithSum(int [] array,int sum) {
        int i = 0;
        int j = array.length - 1;
        ArrayList<Integer> res = new ArrayList<Integer>();
        while(i < j){
            if(array[i] + array[j] == sum){
                res.add(array[i]);
                res.add(array[j]);
                break;
            }
            while(i < j && array[i] + array[j] > sum)
                j--;
            while(i < j && array[i] + array[j] < sum)
                i++;
        }
        return res;
    }
}